src/WaitingRoom.js

```diff
@@ -49,7 +57,16 @@ export default class WaitingRoom extends Component {
       )
     }
 
-    var cards = this.state.sessions.map((session) => (
+    let sessions = [];
+    if(this.state.filterType === "all") {
```

Comment:
Use a switch statement instead of if-else blocks:

```
switch (this.state.filterType) {
  case FILTER_TYPE.ALL:
    ...
    break;
  case FILTER_TYPE.MY_GAMES:
    ...
    break;
  default: // Log an error.
}
```

Even better, you could create a `getFilterFunction` method that returns the function to pass into `Array.filter`, e.g.:

```
getFilterFunction() {
  switch (this.state.filterType) {
    case FILTER_TYPE.ALL:
      return (session) => true; // no-op
      break;
    case FILTER_TYPE.MY_GAMES:
      return (session) => session.creator === firebase.auth().currentUser.uid;
      break;
    default: // Log an error.
  }
}

...

var sessions = this.state.sessions.filter(this.getFilterFunction());
```

Reply:
Can do!